index.js

```diff
@@ -40,6 +34,35 @@ module.exports = function (src, dest, opts) {
 		read.on('end', function () {
 			resolve(read);
 		});
+	});
+};
+
+function statSource(src, progress) {
+	if (progress.enabled) {
+		return fsP.stat(src).then(function(stat) {
+			progress.stat = stat;
```

Comment:
Side effects are hard to understand, why not just return `stat`? (i.e. drop the whole `.then(…)`)